docs/lang/articles/type_system/type.md

```diff
@@ -5,21 +5,37 @@ sidebar_position: 1
 
 # Type system
 
-Taichi supports two data types in the [Taichi scope](../kernels/syntax.md#taichi-scope): primitive type and compound type.
 
-- Primitive types: Various commonly-used numerical data types, such as `int32`, `uint8`, and `float64`
-- Compound types: User-defined data types, which comprise multiple members.
+Taichi is a statically typed programming language. The type of a variable in the Taichi scope is determined at compile time. Once a variable is declared, you cannot bind its name to a value of a different type afterward.
```

Comment:
```suggestion
Taichi is a statically typed programming language. The type of a variable in the Taichi scope is determined at compile time. After a variable is declared, you cannot assign to it a value of a different type.
```

I don't really understand what "bind its name to a value of a different type" means :-)